crates/wasi-http/src/types_impl.rs

```diff
@@ -123,7 +123,13 @@ impl crate::types::Host for WasiHttp {
         bail!("unimplemented: drop_incoming_request")
     }
     fn drop_outgoing_request(&mut self, request: OutgoingRequest) -> wasmtime::Result<()> {
-        self.requests.remove(&request);
+        match self.requests.get(&request) {
```

Comment:
nit; since we only care about the `Some` case, how about an `if let Some(r) = self.requests.get(&request)` instead?

Or even
```
if let Entry::Occupied(e) = self.requests.entry(request) {
   let r = e.remove();
```

https://doc.rust-lang.org/std/collections/hash_map/struct.OccupiedEntry.html#method.remove

Reply:
Done.

Thanks for the suggestion, I'm still learning idiomatic rust)